earthfile2llb/wait_block.go

```diff
@@ -0,0 +1,194 @@
+package earthfile2llb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/earthly/earthly/states"
+	"github.com/earthly/earthly/util/llbutil"
+	"github.com/earthly/earthly/util/llbutil/pllb"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+	"github.com/moby/buildkit/frontend/gateway/client"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/pkg/errors"
+)
+
+type saveImagesWaitItem struct {
+	converters []*Converter
+	images     []states.SaveImage
+	push       []bool
```

Comment:
Make a struct out of these so that there is a single slice?

```golang
type saveImagesWaitItem struct {
	images []saveImage
}

type saveImage struct {
	converter  *Converter
	image      states.SaveImage
	push       bool
}
```